src/dataframe/dataframe.jl

```diff
@@ -793,18 +793,40 @@ julia> deleteat!(df, 2)
    2 │     3      6
 ```
 """
-function Base.deleteat!(df::DataFrame, inds)
-    if !isempty(inds) && size(df, 2) == 0
+Base.deleteat!(df::DataFrame, ::Colon) = empty!(df)
+
+# Bool is accepted here because it is accepted in Base Julia
+function Base.deleteat!(df::DataFrame, inds::Integer)
+    size(df, 2) == 0 && throw(BoundsError(df, (inds, :)))
+    return _deleteat!_helper(df, Int[inds])
+end
+
+function Base.deleteat!(df::DataFrame, inds::AbstractVector)
+    if isempty(inds)
+        return df
+    elseif size(df, 2) == 0
         throw(BoundsError(df, (inds, :)))
     end
 
+    if Bool <: eltype(inds) && any(x -> x isa Bool, inds)
+        throw(ArgumentError("invalid index of type Bool"))
+    end
+
+    if !(eltype(inds) <: Integer || all(x -> x isa Integer, inds))
```

Comment:
Same here?
```suggestion
     # this is required because of https://github.com/JuliaData/InvertedIndices.jl/issues/31
    if !(eltype(inds) <: Integer || all(x -> x isa Integer, inds))
```

Reply:
No, here it is "just required". Here we know that `inds` is `AbstractVector` so it is not connected with `Not`. We just need to make sure all elements are `Integer`, as otherwise in line 826 in general user could get a weird error that `<=` is not defined for values passed.

I know it adds some performance overhead, but this method anyway is not type stable, so I think it is OK to add this check.